pkg/odo/cli/component/create.go

```diff
@@ -359,8 +390,20 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 		co.appName = genericclioptions.ResolveAppFlag(cmd)
 
+		var catalogList catalog.ComponentTypeList
+		if co.forceS2i {
+			client := co.Client
+			catalogList, err = catalog.ListComponents(client)
+			if err != nil {
+				return err
+			}
+		}
+
 		// Validate user specify devfile path
 		if co.devfileMetadata.devfilePath.value != "" {
+			if co.forceS2i {
+				return errors.New("You can't set --s2i flag as true if you want to use the devfile that is specified via --devfile")
```

Comment:
errors cannot be capitalized. Please remove the capitalization at the beginning.